projects/packages/backup-helper-script-manager/src/class-throw-on-errors.php

```diff
@@ -0,0 +1,497 @@
+
+	/**
+	 * Remove a directory, throw on warnings / errors.
+	 *
+	 * @see https://php-legacy-docs.zend.com/manual/php5/en/function.rmdir
+	 * @see https://www.php.net/manual/en/function.rmdir.php
+	 *
+	 * @param string $directory Directory path.
+	 *
+	 * @return void
+	 * @throws Exception On invalid parameters, if rmdir() has thrown warnings, or has failed.
+	 */
+	public static function t_rmdir( $directory ) {
+		// PHP 5.x won't complain about parameter being unset, so let's do it ourselves.
+		if ( ! $directory ) {
+			throw new Exception( 'Directory for mkdir() is unset' );
+		}
+
+		$label = "rmdir( '$directory' )";
+
+		$rmdir_result = static::throw_on_warnings(
+			function () use ( $directory ) {
+				// phpcs:ignore WordPress.WP.AlternativeFunctions.file_system_operations_rmdir
+				return rmdir( $directory );
+			},
+			$label
+		);
+
+		if ( false === $rmdir_result ) {
+			throw new Exception( "Unable to $label" );
+		}
+	}
+
+	/**
+	 * Delete a file, throw on warnings / errors.
+	 *
+	 * @see https://php-legacy-docs.zend.com/manual/php5/en/function.unlink
+	 * @see https://www.php.net/manual/en/function.unlink.php
+	 *
+	 * @param string $filename Path to the file.
+	 *
+	 * @return void
+	 * @throws Exception If unlink() has thrown warnings, or has failed.
+	 */
+	public static function t_unlink( $filename ) {
+
+		$label = "unlink( '$filename' )";
+
+		$unlink_result = static::throw_on_warnings(
+			function () use ( $filename ) {
+				// phpcs:ignore WordPress.WP.AlternativeFunctions.unlink_unlink
+				return unlink( $filename );
+			},
+			$label
+		);
+
+		if ( false === $unlink_result ) {
+			throw new Exception( "Unable to $label" );
+		}
+	}
+
+	/**
+	 * Write data to a file, throw on warnings / errors.
+	 *
+	 * @see https://php-legacy-docs.zend.com/manual/php5/en/function.file-put-contents
+	 * @see https://www.php.net/manual/en/function.file-put-contents.php
+	 *
+	 * @param string $filename Path to the file where to write the data.
+	 * @param string $data The data to write.
+	 *
+	 * @return int Number of bytes that were written to the file.
+	 * @throws Exception If file_put_contents() has thrown warnings, or has failed.
+	 */
+	public static function t_file_put_contents( $filename, $data ) {
+
+		// PHP 5.x won't complain about parameter being unset, so let's do it ourselves.
+		if ( ! $filename ) {
+			throw new Exception( 'Filename for file_put_contents() is unset' );
+		}
+		if ( $data === null ) {
+			throw new Exception( 'Data to write is null' );
+		}
+
+		$label = "file_put_contents( '$filename', " . strlen( $data ) . ' bytes of data )';
+
+		$file_put_contents_result = static::throw_on_warnings(
+			function () use ( $filename, $data ) {
+				// phpcs:ignore WordPress.WP.AlternativeFunctions.file_system_operations_file_put_contents
+				return file_put_contents( $filename, $data );
+			},
+			$label
+		);
+
+		if ( false === $file_put_contents_result ) {
+			throw new Exception( "Unable to $label" );
+		}
```

Comment:
No check for `$file_put_contents_result < strlen( $data )`?

Reply:
Thanks, I've assumed that somehow `file_put_contents()` would bother enough to throw a warning if it doesn't manage to write everything, but that might not always be the case.

Added a check in 883a2c9.